app/components/common/pagination.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+  <nav v-if="props.recordCount" aria-label="{{ t('pagination') }}" class="pagination d-flex justify-content-center text-center">
+    <ul class="pagination">
+
+      <li v-if="props.firstLastButton" @click="setPage(1)" :class="{ 'disabled': internalCurrentPage === 1 }" class="page-item">
+        <a class="page-link">{{ props.firstButtonText }}</a>
+      </li>
+      
+      <li @click="previousPage" :class="{ 'disabled': internalCurrentPage === 1 }" class="page-item">
+        <a class="page-link">{{ props.prevText }}</a>
+      </li>
+      
+      <li v-if="visiblePageNumbers[0] > 1" class="page-item">
+        <a class="page-link">...</a>
+      </li>
+
+      <li v-for="pageNumber in visiblePageNumbers" :key="pageNumber" @click="setPage(pageNumber)" :class="{ 'active': internalCurrentPage === pageNumber }" class="page-item">
+        <a class="page-link">{{ pageNumber }}</a>
+      </li>
+
+      <li v-if="visiblePageNumbers[visiblePageNumbers.length - 1] < pageCount" class="page-item">
+        <a class="page-link">...</a>
+      </li>
+
+      <li @click="nextPage" :class="{ 'disabled': internalCurrentPage === pageCount }" class="page-item">
+        <a class="page-link">{{ props.nextText }}</a>
+      </li>
+
+      <li v-if="props.lastButtonText" @click="setPage(pageCount)" :class="{ 'disabled': internalCurrentPage === pageCount }" class="page-item">
+        <a class="page-link">{{ props.lastButtonText }}</a>
+      </li>
+
+    </ul>
+  </nav>
+
+</template>
+
+<script setup>
+import { ref, computed, defineProps, defineEmits } from 'vue';
+import { useI18n } from 'vue-i18n';
+import messages from '../../app-text/components/common/pagination.json';
+const { t } = useI18n({ messages });
+
+const props = defineProps({
+  recordCount: { type: Number, required: true },
+  recordsPerPage: { type: Number, required: true },
+  currentPage: { type: Number, required: true },
+  firstLastButton: { type: Boolean, default: true },
+  firstButtonText: { type: String, default: 'First' },
+  lastButtonText: { type: String, default: 'Last' },
+  prevText: { type: String, default: 'Prev' },
+  nextText: { type: String, default: 'Next' }
+});
+
+const emit = defineEmits(['changePage']);
+
+const internalCurrentPage = ref(props.currentPage);
+
+const setPage = (page) => {
+  internalCurrentPage.value = page;
+  emit('changePage', page);
+};
```

Comment:
is there a reason to have internalCurrentPageif the value is pushed to the parent `v-model:current-page`?
why not make it computer with setter? make be it was not designed this way.
eg:
```suggestion
const internalCurrentPage = computed({
  get() { return props.currentPage; }
  set(page) { emit('update:changePage', page); }
});ref(props.currentPage);
```

Reply:
use computed properties.